pulsar-client/src/main/java/org/apache/pulsar/client/impl/MultiTopicsConsumerImpl.java

```diff
@@ -571,6 +571,11 @@ protected CompletableFuture<Void> doReconsumeLater(Message<?> message, AckType a
 
     @Override
     public void negativeAcknowledge(MessageId messageId) {
+        if (getState() != State.Ready) {
+            log.warn("[{}] [{}] Cannot negative acknowledge message {} - consumer is not ready (state: {})",
```

Comment:
Would it be better to use `throw new PulsarClientException("Consumer is not ready: " + getStats().toString())` instead?

I'm concerned that logging a warning might give users the impression that the issue is not critical(I know the consumer may be closed), we should handle it explicitly.

Reply:
@nodece 

It will break the API since we didn't declare the PulsarClientException for the method signature.

And if the consumer gets closed, all the unacked messages will be redelivered automatically. Users don't need to care about the failures of negativeAcknowledge() in this case.